android/src/main/java/com/robinpowered/react/ScreenBrightness/ScreenBrightnessModule.java

```diff
@@ -84,20 +90,20 @@ public void onPermissionResult() {
     private boolean hasSettingsPermission() {
         return Build.VERSION.SDK_INT < Build.VERSION_CODES.M ||
                 (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M &&
-                Settings.System.canWrite(getReactApplicationContext()));
+                        Settings.System.canWrite(getReactApplicationContext()));
     }
 
     /**
      * Invokes the request permission activity to request access for WRITE_SETTINGS.
      */
     private void requestSettingsPermission() {
-        Context context = getReactApplicationContext();
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.System.canWrite(context)) {
+        ReactApplicationContext application = getReactApplicationContext();
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && !Settings.System.canWrite(application)) {
             Intent intent = new Intent(
                     Settings.ACTION_MANAGE_WRITE_SETTINGS,
-                    Uri.parse("package:" + context.getPackageName())
+                    Uri.parse("package:" + application.getPackageName())
             );
-            activity.startActivityForResult(intent, writeSettingsRequestCode);
+            application.startActivityForResult(intent, writeSettingsRequestCode, null);
```

Comment:
You might be able to simplify some of this with `ActivityCompat.requestPermissions`

https://github.com/robinpowered/robin-compass/blob/develop/android/app/src/main/java/com/robinpowered/compass/PermissionRequestActivity.java#L31